source/elements/oneMKL/source/domains/rng/device_routines/device-rng-usage-model.rst

```diff
@@ -0,0 +1,110 @@
+.. SPDX-FileCopyrightText: 2023 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_device_rng_usage_model:
+
+oneMKL RNG Device Usage Model
+=============================
+
+.. contents::
+    :local:
+    :depth: 1
+
+A typical usage model for device routines is the same as described in
+:ref:`onemkl_rng_usage_model`:
+
+
+#. Create and initialize the object for basic random number generator.
+
+#. Create and initialize the object for distribution generator.
+
+#. Call the generate routine to get random numbers with appropriate statistical distribution.
+
+
+Example of Scalar Random Numbers Generation
+-------------------------------------------
+
+.. code-block:: cpp
+
+    #include <iostream>
+    #include <sycl/sycl.hpp>
+
+    #include "oneapi/mkl/rng/device.hpp"
+
+    int main() {
+        sycl::queue queue;
+        const int n = 1000;
+        const int seed = 1;
+        // Prepare an array for random numbers
+        std::vector<float> r(n);
+
+        sycl::buffer<float, 1> r_buf(r.data(), r.size());
```

Comment:
Can we remove 1 in sycl::buffer template parameters?

Reply:
I'd also replace it with USM. The example can be taken from the following article: https://www.intel.com/content/www/us/en/developer/articles/technical/random-number-generation-curand-and-onemkl.html

Reply:
removed, thanks

Reply:
My question is still actual

Reply:
Examples changed. Thanks for your help!